### PR TITLE
Changelog and docs for 2.0.5-rc1

### PR DESCRIPTION
**Pull request by clumens**
This is for after master gets merged into the 2.0 branch.  I'm especially curious to know if I did the soname stuff right.

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,3 +1,75 @@
+* Thu Oct 15 2020 Chris Lumens <clumens@redhat.com> Pacemaker-2.0.5
+- Changesets: 497
+- Diff: 285 files changed, 22704 insertions(+), 14458 deletions(-)
+
+- Features added since Pacemaker-2.0.4
+  + cibsecret: read value to set from input (or stdin) if not specified
+  + controller: add new IPC API command for getting node list
+  + libpe_status: Add type="integer" to rule elements, allowing for
+				  specifying 64-bit integers.
```
Comment (kgaillot) [inline ChangeLog:9]: replace libpe_status/libpe_rules/pengine/pe etc. with "scheduler:"

The release notes should be more consistent than developers :)

File: ChangeLog [hunk 2]
```diff
@@ -1,3 +1,75 @@
+* Thu Oct 15 2020 Chris Lumens <clumens@redhat.com> Pacemaker-2.0.5
+- Changesets: 497
+- Diff: 285 files changed, 22704 insertions(+), 14458 deletions(-)
+
+- Features added since Pacemaker-2.0.4
+  + cibsecret: read value to set from input (or stdin) if not specified
+  + controller: add new IPC API command for getting node list
```
Comment (kgaillot) [inline ChangeLog:7]: API isn't considered features of user interest, so we don't put them in the changelog.

However I have been thinking we should start documenting public API changes in a release, probably a new section just before "Changes". I wouldn't bother this time unless you want to go through all the public API changes since 2.0.4 (per your soname changes, libcrmcommon, libpe_status and libpe_rules changed).

File: ChangeLog [hunk 3]
```diff
@@ -1,3 +1,75 @@
+* Thu Oct 15 2020 Chris Lumens <clumens@redhat.com> Pacemaker-2.0.5
+- Changesets: 497
+- Diff: 285 files changed, 22704 insertions(+), 14458 deletions(-)
+
+- Features added since Pacemaker-2.0.4
+  + cibsecret: read value to set from input (or stdin) if not specified
```
Comment (kgaillot) [inline ChangeLog:6]: I would reword this to put cibsecret in the description and head it "tools:" (and move it with the rest of the new tool features)

File: ChangeLog [hunk 4]
```diff
@@ -1,3 +1,75 @@
...
+  + controller: add new IPC API command for getting node list
+  + libpe_status: Add type="integer" to rule elements, allowing for
+				  specifying 64-bit integers.
+  + scheduler: Add rule-based tests to op_defaults and rsc_defaults.
+  + scheduler: Add on-fail=demote and no-quorum-policy=demote recovery
+               policies for promoted resources.
+  + tools: Add --resource= to filter crm_mon output for a resource.
+  + tools: Add -P to crmadmin to ping via pacemakerd-api.
```
Comment (kgaillot) [inline ChangeLog:14]: something like "... to show pacemakerd status" would be more user-friendly (though still somewhat obscure ...)

File: ChangeLog [hunk 5]
```diff
@@ -1,3 +1,75 @@
...
+  + tools: Use glib for cmdline handling in crmadmin and crm_resource.
```
Comment (kgaillot) [inline ChangeLog:15]: This wouldn't go in the changelog unless there was some specific new behavior that you wanted to call out

File: ChangeLog [hunk 6]
```diff
@@ -1,3 +1,75 @@
...
+
+- Fixes for regressions introduced in Pacemaker-2.0.4
+  + scheduler: Add the node name back to bundle instances.
```
Comment (kgaillot) [inline ChangeLog:18]: This is user-visible only in crm_mon, so I'd move it to "tools:" and mention crm_mon (or something like "status displays" or "crm_mon and other tools" if you want to be more general to include crm_resource etc.)

File: ChangeLog [hunk 7]
```diff
@@ -1,3 +1,75 @@
...
+  + tools: get stonith_admin --list-targets working again
+
+- Fixes for regressions introduced in Pacemaker-2.0.3
+  + scheduler: Fix adding HTTP headers to crm_mon in daemon mode.
```
Comment (kgaillot) [inline ChangeLog:22]: I'd put this as "tools:"

File: ChangeLog [hunk 8]
```diff
@@ -1,3 +1,75 @@
...
+
+- Fixes for regressions introduced in Pacemaker-2.0.1
+  + scheduler: require pre-/post-start notifications correctly.
```
Comment (kgaillot) [inline ChangeLog:25]: BTW I wish I had a better description than this, but I'm not sure what situations would trigger the issue

File: ChangeLog [hunk 9]
```diff
@@ -1,3 +1,75 @@
...
+
+- Changes since Pacemaker-2.0.4
+  + CIB: use crm_exit to free qb-logging
```
Comment (kgaillot) [inline ChangeLog:28]: Things that are invisible to users can be left out the changelog

File: ChangeLog [hunk 10]
```diff
@@ -1,3 +1,75 @@
...
+  + cibsecret: don't use pssh -q option unless supported
+  + controller: Log correct timeout for timed-out stonith monitor
+  + controller: properly detect remote node info requests
```
Comment (kgaillot) [inline ChangeLog:31]: This was a fix within this release and so doesn't need to be mentioned (actually it was a regression in 2.0.0, but it wasn't significant until a change in this release)

File: ChangeLog [hunk 11]
```diff
@@ -1,3 +1,75 @@
...
+  + daemons: Accept PCMK_OCF_DEGRADED and PCMK_OCF_DEGRADED_MASTER status codes
```
Comment (kgaillot) [inline ChangeLog:32]: This was a fix, but I think it might be significant enough to list as a feature, since it was an intended feature that never worked before. I'd reword it along the lines of "Recognize new OCF agent status codes 190 (degraded) and 191 (degraded master) to be treated as success but displayed as errors"

File: ChangeLog [hunk 12]
```diff
@@ -1,3 +1,75 @@
...
+  + daemons: Notify tengine of completion of CRM_OP_LRM_DELETE.(CLBZ:#5424)
```
Comment (kgaillot) [inline ChangeLog:33]: Something like "controller: avoid recovery delay when shutdown locks expire"

FYI it's very difficult to write changelog entries without having been involved in most of the PRs. Developers naturally write commit messages about implementation, whereas the changelog needs to have the user-visible effects. Sometimes the full commit message gives enough context, otherwise sometimes the PR description does, and it's usually not worth the time chasing it down beyond that.

File: ChangeLog [hunk 13]
```diff
@@ -1,3 +1,75 @@
...
+  + daemons: Remove relayed stonith operation.(CLBZ#5401)
```
Comment (kgaillot) [inline ChangeLog:34]: "tools: avoid pending fence actions getting stuck in crm_mon display"

File: ChangeLog [hunk 14]
```diff
@@ -1,3 +1,75 @@
...
+  + executor: only send executor notifications to executor clients
```
Comment (kgaillot) [inline ChangeLog:35]: "tools: avoid crm_node on Pacemaker Remote nodes failing when other executor actions are occurring"

File: ChangeLog [hunk 15]
```diff
@@ -1,3 +1,75 @@
...
+  + fencer:  Add an additional check to advance_topology_level.
```
Comment (kgaillot) [inline ChangeLog:36]: This was just a static analysis thing, not worth an entry

File: ChangeLog [hunk 16]
```diff
@@ -1,3 +1,75 @@
...
+  + fencer: avoid infinite loop if device is removed during operation
+  + fencer: avoid possible use-of-NULL when parsing metadata
+  + ipc-api: allow calling connect after disconnection
+  + ipc-api: avoid infinite loop when disconnected
```
Comment (kgaillot) [inline ChangeLog:40]: I'm pretty sure the above two "ipc-api" entries were for issues within the release, so no entry needed

File: ChangeLog [hunk 17]
```diff
@@ -1,3 +1,75 @@
...
+  + libCIB: Add `tags` and `alerts` as known CIB paths, remove `domains`.
```
Comment (kgaillot) [inline ChangeLog:41]: "tools: allow tags and alerts in cibadmin --scope"

File: ChangeLog [hunk 18]
```diff
@@ -1,3 +1,75 @@
...
+  + libcrmcommon: Allow default formatted output message handlers.
+  + libcrmcommon: Don't write out name="" as part of XML output.
+  + libcrmcommon: Rewrite parse_op_key to understand migrate_to and migrate_from.
+  + libcrmcommon: use correct sense of string comparison in crm_action_str()
+  + libcrmcommon: Use truly implicit deny for ACLs
```
Comment (kgaillot) [inline ChangeLog:46]: all of the above are user-invisible implementation details

File: ChangeLog [hunk 19]
```diff
@@ -1,3 +1,75 @@
...
+  + libfencing: add `port` or `plug` parameter according to metadata on
+                `validate` if no `pcmk_host_argument` specified
+  + libfencing: respect `pcmk_host_argument=none` on `validate`
+  + libpacemaker: Don't allow a potential NULL in a format string.
```
Comment (kgaillot) [inline ChangeLog:50]: This was a build issue in Fedora 32 on s390, probably not worth an entry, otherwise "fix build on ..."

File: ChangeLog [hunk 20]
```diff
@@ -1,3 +1,75 @@
...
+  + libpe_status: Don't output unnecessary resource headers.
+  + libpe_status: Handle filtering of inactive resources.
```
Comment (kgaillot) [inline ChangeLog:52]: the above two probably aren't worth mentioning

File: ChangeLog [hunk 21]
```diff
@@ -1,3 +1,75 @@
...
+  + libpe_status: Support integer as rule type attribute
```
Comment (kgaillot) [inline ChangeLog:53]: this should be moved up to features and maybe relabeled "configuration:"

File: ChangeLog [hunk 22]
```diff
@@ -1,3 +1,75 @@
...
+  + libpe_status: Use pcmk_monitor_timeout as stonith start timeout
```
Comment (kgaillot) [inline ChangeLog:54]: s/libpe_status/scheduler/

File: ChangeLog [hunk 23]
```diff
@@ -1,3 +1,75 @@
...
+  + pacemakerd: improve logging when synced with SBD
+  + sbd-integration: sync pacemakerd with sbd
```
Comment (kgaillot) [inline ChangeLog:56]: This should be under features as "support SBD_SYNC_RESOURCE_STARTUP environment variable to better synchronize Pacemaker start-up and shutdown with SBD"

File: ChangeLog [hunk 24]
```diff
@@ -1,3 +1,75 @@
...
+  + scheduler: Add management status to group output.
+  + scheduler: If a clone or group is disabled, display that in the rsc header.
```
Comment (kgaillot) [inline ChangeLog:58]: the above two should be "tools:" and mention crm_mon

File: ChangeLog [hunk 25]
```diff
@@ -1,3 +1,75 @@
...
+  + scheduler: disallow on-fail=stop for stop operations
+  + scheduler: Display stop-all-resources in cluster options.
```
Comment (kgaillot) [inline ChangeLog:60]: also tools/crm_mon

File: ChangeLog [hunk 26]
```diff
@@ -1,3 +1,75 @@
...
+  + scheduler: don't schedule a dangling migration stop if one already occurred
+  + scheduler: don't select instance to be promoted on a guest that can't run
+  + tools, libcrmcommon: Reopen the output dest on reset.
```
Comment (kgaillot) [inline ChangeLog:63]: not needed

File: ChangeLog [hunk 27]
```diff
@@ -1,3 +1,75 @@
...
+  + tools: Add "No active resources" to one case in crm_mon output.
+  + tools: Fix addding the http-equiv header to crm_mon in daemon mode.
+  + tools: Change return codes of cli_resource_execute* functions.
```
Comment (kgaillot) [inline ChangeLog:66]: not needed

File: ChangeLog [hunk 28]
```diff
@@ -1,3 +1,75 @@
...
+  + tools: crm_node -l and -p now work from Pacemaker Remote nodes
+  + tools: Don't display crm_resource error messages twice.
+  + tools: Print inactive resources by default with crm_mon xml output.
+  + tools: properly detect local node name
```
Comment (kgaillot) [inline ChangeLog:70]: in cibsecret

File: ChangeLog [hunk 29]
```diff
@@ -1,3 +1,75 @@
...
+  + tools: Use standard pacemaker return codes in crm_resource.
```
Comment (kgaillot) [inline ChangeLog:71]: not needed

**Review by kgaillot — CHANGES_REQUESTED** [review]
soname stuff looks good

File: ChangeLog (hunk 2 above)
Comment (clumens) [reply ChangeLog:7]: 2.1.0 seems like a great time to start doing this.

File: ChangeLog (hunk 5 above)
Comment (clumens) [reply ChangeLog:15]: It was in at least one previous changelog entry, which is why I left it.  But I'll get rid of it here.

File: ChangeLog (hunk 9 above)
Comment (clumens) [reply ChangeLog:28]: Yeah, I figured.  I tried to get rid of as many of those as I could.  This one just slipped through.

File: ChangeLog (hunk 19 above)
Comment (clumens) [reply ChangeLog:50]: Changed to explain it as a build problem.

File: ChangeLog (hunk 18 above)
Comment (clumens) [reply ChangeLog:46]: I am unsure if anyone would have ever seen the first one.  Probably not.  I'll get rid of all four.

File: ChangeLog [hunk 30]
```diff
@@ -1,3 +1,65 @@
...
+- Features added since Pacemaker-2.0.4
+  + configuration: Support integer as rule type attribute.
+  + daemons: Recognize new OCF agent status codes 190 (degraded) and 191
+             (degraded master) to be treated as success but displayed as
+             errors.
+  + sbd-integration: support SBD_SYNC_RESOURCE_STARTUP environment
+                     variable to better synchronize Pacemaker start-up
+                     and shutdown with SBD
+  + scheduler: Add type="integer" to rule elements, allowing for
+               specifying 64-bit integers.
```
Comment (kgaillot) [inline ChangeLog:14]: I just noticed we have the integer thing in there twice

Also, add something like "and specifying double-precision floating-point numbers when type="number"" (number was formerly handled as 32-bit integer)

File: ChangeLog [hunk 31]
```diff
@@ -1,3 +1,65 @@
...
+  + cibsecret: properly detect local node name
+  + controller: avoid recovery delay when shutdown locks expire
+  + controller: Log correct timeout for timed-out stonith monitor
+  + fencer: avoid infinite loop if device is removed during operation
+  + fencer: avoid possible use-of-NULL when parsing metadata
+  + libcrmcommon: Allow default formatted output message handlers.
+  + libfencing: add `port` or `plug` parameter according to metadata on
+                `validate` if no `pcmk_host_argument` specified
+  + libfencing: respect `pcmk_host_argument=none` on `validate`
+  + pacemakerd: improve logging when synced with SBD
```
Comment (kgaillot) [inline ChangeLog:44]: We can drop this one, it was part of the sync feature mentioned earlier

File: ChangeLog [hunk 32]
```diff
@@ -1,3 +1,65 @@
...
+- Fixes for regressions introduced in Pacemaker-2.0.3
+  + tools: Fix adding HTTP headers to crm_mon in daemon mode.
+
+- Fixes for regressions introduced in Pacemaker-2.0.1
+  + scheduler: require pre-/post-start notifications correctly.
+
+- Changes since Pacemaker-2.0.4
+  + build: Fix a build issue on Fedora s390x.
+  + cibsecret: don't use pssh -q option unless supported
+  + cibsecret: properly detect local node name
```
Comment (kgaillot) [inline ChangeLog:35]: the above two can be "tools:"

File: ChangeLog [hunk 33]
```diff
@@ -1,3 +1,65 @@
...
+  + controller: avoid recovery delay when shutdown locks expire
+  + controller: Log correct timeout for timed-out stonith monitor
+  + fencer: avoid infinite loop if device is removed during operation
+  + fencer: avoid possible use-of-NULL when parsing metadata
+  + libcrmcommon: Allow default formatted output message handlers.
```
Comment (kgaillot) [inline ChangeLog:40]: not needed

File: ChangeLog [hunk 34]
```diff
@@ -1,3 +1,65 @@
...
+  + scheduler: don't select instance to be promoted on a guest that can't run
+  + scheduler: Use pcmk_monitor_timeout as stonith start timeout
+  + tools: Add management status to crm_mon's group output.
+  + tools: Add "No active resources" to one case in crm_mon output.
+  + tools: Allow tags and alerts in cibadmin --scope
+  + tools: Avoid crm_node on Pacemaker Remote nodes failing when other
+           executor actions are occurring.
+  + tools: Avoid pending fence actions getting stuck in crm_mon display
+  + tools: Display stop-all-resources in crm_mon's cluster options.
+  + tools: Fix addding the http-equiv header to crm_mon in daemon mode.
```
Comment (kgaillot) [inline ChangeLog:56]: spelling

**Review by kgaillot — CHANGES_REQUESTED** [review]
Looking better. We've got a nice set of changes this release ...

**Comment by kgaillot** [comment]
Looks good to me, pull the trigger when ready
